### PR TITLE
Added canary deployment and switch

### LINKED ISSUE #64: Canary deployments and roll back
## Feature description
Opportunity  for canary deployments and roll back

## Why is it needed?
Improve resilience 

## Suggestion(s)/solution(s) [Optional]
Add `update` stanza to nomad job

## Definition of done
Dynamically setup use of canary and roll back feature

## Checklist (after created issue)
- [x] Added label(s)
- [x] Added to project
- [x] Added to milestone

### PR DESCRIPTION
**Pull request by dangernil**
## Closes/fixes/resolves issue(s)?
Closes #64 

## What was added/changed/fixed?
Added canary deployment and switch

## Checklist (after created PR)
- [x] Added reviewers
- [x] Added assignee(s)
- [x] Added label(s)
- [x] Added to project
- [x] Added to milestone

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,6 +65,7 @@ Minio example instance has:
 | minio\_service\_name | Minio service name | string |
 | minio\_access\_key | Minio access key | string |
 | minio\_secret\_key | Minio secret key | string |
+| use_canary | Minio canary deployment | bool |
```
Comment (claesgill) [inline README.md]: This should be moved to the *Input* section.

File: conf/nomad/minio.hcl [hunk 2]
```diff
@@ -4,6 +4,18 @@ job "${service_name}" {
...
 
+  update {
+    max_parallel      = 1
+    health_check      = "checks"
+    min_healthy_time  = "10s"
+    healthy_deadline  = "10m"
+    progress_deadline = "12m"
+    auto_revert       = true
+    auto_promote      = true
+    canary            = 1
```
Comment (claesgill) [inline conf/nomad/minio.hcl]: Need to add switch here

File: example/main.tf [hunk 3]
```diff
@@ -20,6 +20,7 @@ module "minio" {
   data_dir                        = "/minio/data"
   container_environment_variables = ["SOME_VAR_N1=some-value"]
   use_host_volume                 = true
+  use_canary                      = false
```
Comment (zhenik) [inline example/main.tf]: Set on the full set of optional features in the example.
```suggestion
  use_canary                      = true
```

File: README.md [hunk 4]
```diff
@@ -65,6 +65,7 @@ Minio example instance has:
...
 
 ### Example
```
Comment (zhenik) [inline README.md:71]: I would suggest updating doc example accordingly to the new feature

**Review by zhenik — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -57,6 +57,7 @@ Minio example instance has:
 | mc\_container\_image | Minio client docker image | string | "minio/mc:latest" | yes |
 | mc\_container\_environment\_variables | Additional minio client container environment variables | list(string) | [] | no |
 | buckets | List of buckets to create on startup | list(string) | [] | no |
+| use_canary | Minio canary deployment | bool |
```
Comment (claesgill) [inline README.md]: Need to fill out all fields like the rest:  Name, Description, Type, Default, Required

**Review by claesgill — CHANGES_REQUESTED** [review]

**Review by claesgill — DISMISSED** [review]
LGTM

File: README.md (hunk 5 above)
Comment (zhenik) [inline README.md]: Add escape char for consistency
```suggestion
| use\_canary | Minio canary deployment | bool |
```

**Review by zhenik — CHANGES_REQUESTED** [review]

**Review by zhenik — APPROVED** [review]
LGTM

File: README.md (hunk 5 above)
Comment (claesgill) [reply README.md]: Why do we need to escape underscores? 🤷 I don't think these are necessary.

**Review by claesgill — APPROVED** [review]

**Pull request merged**